inngest-core/src/test/kotlin/com/inngest/signingkey/BearerTokenKtTest.kt

```diff
@@ -0,0 +1,47 @@
+package com.inngest.signingkey
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+class BearerTokenKtTest {
+    @Test
+    fun `test invalid signing key format`() {
+        assertFailsWith<InvalidSigningKeyException> {
+            getAuthorizationHeader("signke-prod-blah")
+        }
+    }
+
+    @Test
+    fun `test signing key is not a valid hex string`() {
+        assertFailsWith<InvalidSigningKeyException> {
+            getAuthorizationHeader("signkey-prod-8fjau3mn")
+        }
+    }
+
+    @Test
+    fun `builds authorization header with bearer token for prod environment`() {
+        val authorizationHeader = getAuthorizationHeader("signkey-prod-12345678")
+        assertEquals(
+            "Authorization",
+            authorizationHeader.keys.first(),
+        )
+        assertEquals(
+            "Bearer signkey-prod-b2ed992186a5cb19f6668aade821f502c1d00970dfd0e35128d51bac4649916c",
+            authorizationHeader.values.first(),
+        )
+    }
+
+    @Test
+    fun `builds authorization header with bearer token for non prod environment`() {
+        val authorizationHeader = getAuthorizationHeader("signkey-test-12345678")
+        assertEquals(
+            "Authorization",
+            authorizationHeader.keys.first(),
+        )
+        assertEquals(
+            "Bearer signkey-test-b2ed992186a5cb19f6668aade821f502c1d00970dfd0e35128d51bac4649916c",
```

Comment:
There might be a timestamp validation to the signature though my memory is a little blurry around here.

Reply:
Yeah there is a timestamp. i think he has a second PR coming that includes that.

Reply:
yeah, we realized the signing key ends up having dual use, one for creating a bearer token for requests from SDK to Inngest server, and the other is for validating signatures for requests from Inngest server to the SDK, which is what has the timestamp component (and is coming in my next PR)

Reply:
Sounds good to me.